### DISCUSSION TITLE
workarounds:supportRedHatImageVersion pattern matching broken

### DISCUSSION THREAD

**Discussion by sichapman** [renovatebot/renovate#31314] [Request Help] [answered]
### What would you like help with?

I think I found a bug

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitLab

### Please tell us more about your question or problem

These patterns

```
registry.access.redhat.com/ubi7**
registry.access.redhat.com/ubi8**
registry.access.redhat.com/ubi9**
```

do not match any package name where the first character of the ** is a /

e.g. `registry.access.redhat.com/ubi9/openjdk-11-runtime` - no match
`registry.access.redhat.com/ubi9-minimal`  - match

I've verified this in standalone minimatch so it must just be a quirk of how the matching works.

The three patterns above could be replaced with `registry.access.redhat.com/ubi*/**` and that should ensure all the various ubi variations are matched, as well as future proofing for ubi10. I have tested this change with a manual packagerule in my repo's renovate.json and it works.

**Comment by rarkins** [comment]
`registry.access.redhat.com/ubi*/**` does not match your example `registry.access.redhat.com/ubi9-minimal`. Is that a real example or should we expect all real package names to have a `/`?

**Reply by sichapman** [reply]
Oops, I was juggling too many images and didn't think that solution through! All my examples are real, they have ones with and without a slash after the ubi7/8/9.

**Reply by sichapman** [reply]
I guess two patterns, `registry.access.redhat.com/ubi*` and `registry.access.redhat.com/ubi*/**` could take care of it but I'm no expert (clearly!)

**Reply by rarkins** [reply]
Yes, that could be simplest. There's perhaps an advanced glob pattern which works for both

**Reply by viceice** [reply]
`registry.access.redhat.com/ubi{,/}**` works in both cases

**Reply by sichapman** [reply]
That doesn't seen to match against `registry.access.redhat.com/ubi9/openjdk-11-runtime`.

`registry.access.redhat.com/ubi*{,/}**` perhaps?

**Reply by sichapman** [reply, marked as answer]
Have tested `registry.access.redhat.com/ubi*{,/}**` with every ubi package I can find in Red Hat's registry and it matches them all

**Reply by rarkins** [reply]
Great, would you like to submit a PR to fix it?
